forth.asm

```diff
@@ -152,7 +152,9 @@ dict_entry find, 'FIND', 0x0 ;( str -- str | xt 0|1|-1)
 .findcmploop:
     cmpsb
     jne     .nextword
-    cmp     byte [esi], 0x0 ;end of string?
+    cmp     byte [esi], 0x0 ;end of searchstring?
+    jnz     .findcmploop
+    cmp     byte [edi], 0x0 ;end of found string?
```

Comment:
That will fail if the bits in the two bytes are disjoint.  e.g 01010101 and 10101010.  The `and cl, cl` trick only works if it is the exact same value.  In this case, without seeing the surrounding code, I think this might be your best option.